### PR TITLE
Add group option additional fields

### PR DESCRIPTION
**Pull request by arsengit**
- Added additional select options for grouping (group by metric name, experiment name, run hash, context?) 
- Changed grouping selectOptions items type

### CODE REVIEW

File: aim/web/ui_v2/src/types/services/models/metrics/metricsAppModel.d.ts [hunk 1]
```diff
@@ -116,3 +116,8 @@ export interface IGetPersistIndex {
 }
 
 export type groupNames = 'color' | 'style' | 'chart';
```
Comment (roubkar) [inline aim/web/ui_v2/src/types/services/models/metrics/metricsAppModel.d.ts:118]: @aimhubio/uidevs Let's start types names with uppercase and us type keyword at the end.
e.g.
```js
export type GroupNameType = 'color' | 'style' | 'chart';

export type GroupingSelectOptionType = { ... }; 
```

File: aim/web/ui_v2/src/components/ChartPanel/ChartPanel.tsx [hunk 2]
```diff
@@ -98,6 +98,8 @@ const ChartPanel = React.forwardRef(function ChartPanel(
             {props.data.map((lineChartData, index) => (
               <Grid
                 key={index}
```
Comment (roubkar) [inline aim/web/ui_v2/src/components/ChartPanel/ChartPanel.tsx:100]: shouldn't we change the key logic to avoid drawing an additional first chart?

**Review by roubkar — CHANGES_REQUESTED** [review]

File: aim/web/ui_v2/src/components/ChartPanel/ChartPanel.tsx (hunk 2 above)
Comment (arsengit) [reply aim/web/ui_v2/src/components/ChartPanel/ChartPanel.tsx:100]: Yes We Are, but we need to rerender LineChart when chart grouping state changes. So with the logic we wrote LineChart also rerenders when we changing style and color grouping state. I wrote TODO for this issue.

**Review by roubkar — APPROVED** [review]

**Review by KaroMourad — APPROVED** [review]
lgtm

**Pull request merged**